File: vacation/vacation/VacationUsed.go
package vacation

import (
	"time"
)

type VacationUsed struct{
	EventType string	`json:"eventType" type:"string"`
	TimeStamp string 	`json:"timeStamp" type:"string"`
	Id string `json:"id" type:"string"` 
	StartDate  `json:"startDate" type:""` 
	EndDate  `json:"endDate" type:""` 
	Reason string `json:"reason" type:"string"` 
	UserId string `json:"userId" type:"string"` 
	Days int `json:"days" type:"int"` 
	
}

func NewVacationUsed() *VacationUsed{
	event := &VacationUsed{EventType:"VacationUsed", TimeStamp:time.Now().String()}

	return event
}

File: vacation/src/main/java/human/resource/mgmt/domain/VacationUsed.java
package human.resource.mgmt.domain;

import human.resource.mgmt.domain.*;
import human.resource.mgmt.infra.AbstractEvent;
import java.util.*;
import lombok.*;

@Data
@ToString
public class VacationUsed extends AbstractEvent {

    private String id;
    private Date startDate;
    private Date endDate;
    private String reason;
    private String userId;
    private Integer days;

    public VacationUsed(Vacation aggregate) {
        super(aggregate);
    }

    public VacationUsed() {
        super();
    }
}